libs/wingc/src/nodeof_transform.rs

```diff
@@ -0,0 +1,77 @@
+use crate::{
+	ast::{CalleeKind, Expr, ExprKind, Reference, Symbol, UserDefinedType},
+	diagnostic::WingSpan,
+	fold::{self, Fold},
+};
+
+/// Transforms `nodeof(x)` expressions into `std.Node.of(x)`
+pub struct NodeofTransformer;
+
+impl NodeofTransformer {
+	pub fn new() -> Self {
+		Self
+	}
+}
+
+impl Fold for NodeofTransformer {
+	fn fold_expr(&mut self, expr: Expr) -> Expr {
+		let file_id = &expr.span.file_id;
+
+		let ExprKind::Call { callee, arg_list } = expr.kind else {
+			return fold::fold_expr(self, expr);
+		};
+
+		let arg_list = self.fold_args(arg_list);
+
+		let callee = match callee {
+			CalleeKind::Expr(call_expr) => call_expr,
+			_ => {
+				return Expr {
+					kind: ExprKind::Call { callee, arg_list },
+					id: expr.id,
+					span: expr.span,
+				};
+			}
+		};
+
+		let ExprKind::Reference(Reference::Identifier(Symbol { ref name, .. })) = callee.kind else {
+			return Expr {
+				kind: ExprKind::Call {
+					callee: CalleeKind::Expr(callee),
+					arg_list,
+				},
+				id: expr.id,
+				span: expr.span,
+			};
```

Comment:
For some reason just returning `fold::fold_expr(self, expr)` in some of these early-return branches causes me to get borrow checking errors... I'm not sure if there's a better way to avoid this boilerplate